packages/react-scripts/config/webpackDevServer.config.js

```diff
@@ -60,4 +62,10 @@ module.exports = {
   https: protocol === 'https',
   host: host,
   overlay: false,
+  historyApiFallback: {
+    // Paths with dots should still use the history fallback.
+    // See https://github.com/facebookincubator/create-react-app/issues/387.
+    disableDotRule: true,
+  },
```

Comment:
I think there's a conflict with the proxy string as well. I don't think there should be more than one request for the favicon for a default setup:
<img width="462" alt="switch-routes-default-setup" src="https://cloud.githubusercontent.com/assets/873043/23878650/025e17d0-0849-11e7-9eab-6dca329c25d9.png">